client/landing/stepper/declarative-flow/internals/steps-repository/components/paid-memberships-control/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { Onboard } from '@automattic/data-stores';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import { ChangeEvent } from 'react';
+import FormInputCheckbox from 'calypso/components/forms/form-checkbox';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import InfoPopover from 'calypso/components/info-popover';
+import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
+import type { OnboardSelect } from '@automattic/data-stores';
+import './style.scss';
+
+const PaidMembershipsControl = () => {
+	const { __ } = useI18n();
+	const paidSubscribers = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getPaidSubscribers(),
+		[]
+	);
+	const { setPaidSubscribers, setGoals, resetGoals } = useDispatch( ONBOARD_STORE );
+	const onPaidSubscribersChanged = ( event: ChangeEvent< HTMLInputElement > ) => {
+		const isPaidSelected = !! event?.target.checked;
+		setPaidSubscribers( isPaidSelected );
```

Comment:
Note that unlike before, we are directly fetching and updating the value of paidSubscribers from the redux store. We had been pulling in that value, storing it in local state (useState), and then updating the redux store on form submit. But that seems redundant, and doing it this way allows us to consolidate all the logic for this component within the component, rather than having some in the parent so it can be handled by the form submit method.